File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/java/src/main/java/de/sots/cellarsandcentaurs/domain/service/Creatures.java
package de.sots.cellarsandcentaurs.domain.service;

import de.sots.cellarsandcentaurs.domain.model.Creature;
import de.sots.cellarsandcentaurs.domain.model.CreatureId;
import de.sots.cellarsandcentaurs.domain.model.NoSuchCreatureException;

public interface Creatures {
    void save(Creature creature);
    Creature find(CreatureId id) throws NoSuchCreatureException;
}

File: sample-projects/ruby/lib/de/sots/cellars_and_centaurs/domain/service/creatures.rb
# frozen_string_literal: true

require_relative '../../version'
require_relative '../model/creature'
require_relative '../model/creature_id'
require_relative '../model/errors'

module De::Sots::CellarsAndCentaurs::Domain::Service
  module Creatures
    def save(creature)
      raise NotImplementedError, "#{self.class} must implement save"
    end

    # @raise [NoSuchCreatureException] when the id is unknown
    def find(creature_id)
      raise NotImplementedError, "#{self.class} must implement find"
    end
  end
end
